src/features/myAccounts/hooks/useAllSafes.ts

```diff
@@ -42,17 +42,20 @@ export const _buildSafeItem = (
   walletAddress: string,
   allAdded: AddedSafesState,
   allOwned: AllOwnedSafes,
+  allUndeployed: UndeployedSafesState,
   allVisitedSafes: VisitedSafesState,
   allSafeNames: AddressBookState,
 ): SafeItem => {
   const addedSafe = allAdded[chainId]?.[address]
   const addedSafeOwners = addedSafe?.owners || []
+  const undeployedSafeOwners = allUndeployed[chainId]?.[address]?.props.safeAccountConfig.owners || []
   const isPinned = Boolean(addedSafe) // Pinning a safe means adding it to the added safes storage
 
   // Determine if the user is an owner
   const isOwnerFromAdded = addedSafeOwners.some(({ value }) => sameAddress(walletAddress, value))
+  const isOwnerFromCF = undeployedSafeOwners.some((ownedAddress) => sameAddress(walletAddress, ownedAddress))
   const isOwnedSafe = (allOwned[chainId] || []).includes(address)
-  const isOwned = isOwnedSafe || isOwnerFromAdded
+  const isOwned = isOwnedSafe || isOwnerFromAdded || isOwnerFromCF
```

Comment:
I think `isOwnerFromAdded` is not needed anymore. If it's deployed we check ownership from allOwned, if it's CF we check ownership from allUndeployed, so isOwnerFromAdded is redundant